proxy/src/registry.rs

```diff
@@ -514,13 +514,13 @@ impl Client for Registry {
     ) -> Result<Option<Project>, error::Error> {
         Ok(self
             .client
-            .get_project(project_name.0, org_id.0)
+            .get_project(project_name.clone().0, ProjectDomain::Org(org_id.clone().0))
             .await?
             .map(|project| {
                 let metadata_vec: Vec<u8> = project.metadata.into();
                 Project {
-                    name: ProjectName(project.name),
-                    org_id: Id(project.org_id),
+                    name: project_name.clone(),
+                    org_id: org_id.clone(),
```

Comment:
To be transparent about this bit: we request a project under the given `org_id` domain, and when we get it and go on building this `Project` value, instead of unwrapping the `org_id` domain from the registry Project we just use the input param, as they can _only_ be the same. The same goes for the `project_name`. I would usually avoid this to have one source of truth but in this case, it appears sensible, as it's impossible to request a project under an org X and get back a project use domain is not that same org.